apps/betterangels-backend/hmis/api_bridge.py

```diff
@@ -470,3 +479,188 @@ def update_client_note(
         client_note_input: dict[str, Any],
     ) -> Optional[dict[str, Any]]:
         return self._run_client_note_mutation("update", client_note_input)
+
+
+class HmisApiRestBridge:
+    """Utility class for interfacing with HMIS REST API."""
+
+    def __init__(self, info: strawberry.Info) -> None:
+        self.info = info
+        self.request = self.info.context["request"]
+        self.session = self.request.session
+
+        if HMIS_REST_ENDPOINT is None or HMIS_SESSION_KEY_NAME is None or HMIS_HOST is None:
+            raise Exception("HMIS not configured")
+
+        self.endpoint = HMIS_REST_ENDPOINT
+        self.schema = self._load_graphql_schema()
+        self.session_key = HMIS_SESSION_KEY_NAME
+
+        token = self._get_auth_token()
+        auth_header = {"Authorization": f"Bearer {token}"} if token else {}
+
+        self.headers = {
+            "Accept": "application/json, text/plain, */*",
+            "Connection": "keep-alive",
+            "Host": HMIS_HOST,
+            "User-Agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:144.0) Gecko/20100101 Firefox/144.0",
+            **auth_header,
+        }
+
+    def _load_graphql_schema(self) -> GraphQLSchema:
+        obj = import_string(GRAPHQL_SCHEMA_PATH)
+
+        return getattr(obj, "_schema", obj)  # type: ignore
+
+    def _make_request(
+        self,
+        path: str,
+        body: dict[str, Any],
+        method: HTTPMethod = HTTPMethod.GET,
+        timeout: Optional[float] = None,
+    ) -> dict[str, Any]:
+        request_args = {
+            "url": f"{self.endpoint}{path}",
+            "headers": self.headers,
+            "json": body,
+            "timeout": timeout,
+        }
+
+        if method is HTTPMethod.POST:
```

Comment:
can likely replace this branch with 

resp = requests.request(method, **request_args)